k8s/production/runners/protected/graviton/4/release.yaml

```diff
@@ -0,0 +1,192 @@
+---
+apiVersion: source.toolkit.fluxcd.io/v1
+kind: HelmRepository
+metadata:
+  name: runner-graviton4-prot
+  namespace: gitlab
+spec:
+  interval: 10m
+  url: https://charts.gitlab.io
+
+---
+apiVersion: helm.toolkit.fluxcd.io/v2
+kind: HelmRelease
+metadata:
+  name: runner-graviton4-prot
+  namespace: gitlab
+spec:
+  interval: 10m
+  chart:
+    spec:
+      chart: gitlab-runner
+      version: 0.75.0 # gitlab-runner@17.10.0
+      sourceRef:
+        kind: HelmRepository
+        name: runner-graviton4-prot
+  dependsOn:
+    - name: gitlab
+      namespace: gitlab
+
+  valuesFrom:
+    # See terraform/modules/sentry/sentry.tf
+    - kind: ConfigMap
+      name: gitlab-runner-sentry-config
+      valuesKey: values.yaml
+
+  values:
+    imagePullPolicy: IfNotPresent
+    replicas: 3
+    probeTimeoutSeconds: 70
+
+    gitlabUrl: "https://gitlab.spack.io/"
+    unregisterRunners: true
+    terminationGracePeriodSeconds: 21600 # six hours
+    concurrent: 20
+    checkInterval: 30
+
+    metrics:
+      enabled: true
+
+    rbac:
+      serviceAccountName: runner
+
+    runners:
+      config: |
+        [[runners]]
+          pre_build_script = """
+          echo 'Executing Spack pre-build setup script'
+
+          for cmd in "${PY3:-}" python3 python; do
+            if command -v > /dev/null "$cmd"; then
+              export PY3="$(command -v "$cmd")"
+              break
+            fi
+          done
+
+          if [ -z "${PY3:-}" ]; then
+            echo "Unable to find python3 executable"
+            exit 1
+          fi
+
+          $PY3 -c "import urllib.request;urllib.request.urlretrieve('https://raw.githubusercontent.com/spack/spack-infrastructure/main/scripts/gitlab_runner_pre_build/pre_build.py', 'pre_build.py')"
+          $PY3 pre_build.py > envvars
+
+          . ./envvars
+          rm -f envvars
+          unset GITLAB_OIDC_TOKEN
+          """
+
+          output_limit = 20480
+          environment = ["FF_GITLAB_REGISTRY_HELPER_IMAGE=1"]
+          [runners.kubernetes]
+            helper_image = "registry.gitlab.com/gitlab-org/gitlab-runner/gitlab-runner-helper:arm-latest"
+            privileged = false
+            helper_memory_request = "512M"
+
+            cpu_request = "750m"
+            cpu_request_overwrite_max_allowed = "16"
+            cpu_limit_overwrite_max_allowed = "32"
+
+            memory_request = "2G"
+            memory_request_overwrite_max_allowed = "64G"
+            memory_limit = "64G"
+            memory_limit_overwrite_max_allowed = "64G"
+
+            namespace = "pipeline"
+            poll_timeout = 600  # ten minutes
+            service_account = "runner"
+
+            [runners.kubernetes.affinity]
+              [runners.kubernetes.affinity.node_affinity]
+
+              # Schedule this pod on only graviton 4 nodes
```

Comment:
Assuming this is backwards compatible like graviton 3->2 was, we should also update the graviton3 runners to allow for scheduling on graviton4 (at a lower affinity) [like we do for x86](https://github.com/spack/spack-infrastructure/blob/c3d111fe160a8c9bd52f2d65a5c1b9767ca89fba/k8s/production/runners/public/x86_64/v2/release.yaml#L114-L132).

Reply:
@kwryankrattiger do you happen to know if neoverse v2 is backwards compatible with v1?

FWIW, the UO neoverse_v2 runners are currently **not** cross-tagged as v1.

Either way, this PR will need to be updated to either apply the suggest above, or to remove the neoverse_v1 tag from these new runners.

Reply:
based on our f2f conversation yesterday, I opted to treat graviton3 and graviton4 as disjoint sets, rather than cross-labeling these new runners as graviton3.